cli/start.go

```diff
@@ -58,11 +56,41 @@ func (r *RootCmd) start() *clibase.Cmd {
 				return err
 			}
 
-			build, err := client.CreateWorkspaceBuild(inv.Context(), workspace.ID, codersdk.CreateWorkspaceBuildRequest{
+			req := codersdk.CreateWorkspaceBuildRequest{
 				Transition:          codersdk.WorkspaceTransitionStart,
 				RichParameterValues: buildParameters,
-			})
-			if err != nil {
+				TemplateVersionID:   workspace.LatestBuild.TemplateVersionID,
+			}
+
+			build, err := client.CreateWorkspaceBuild(inv.Context(), workspace.ID, req)
+			// It's possible for a workspace build to fail due to the template requiring starting
+			// workspaces with the active version.
+			if cerr, ok := codersdk.AsError(err); ok && cerr.StatusCode() == http.StatusUnauthorized {
+				template, err := client.Template(inv.Context(), workspace.TemplateID)
+				if err != nil {
+					return xerrors.Errorf("get template: %w", err)
+				}
+
+				buildParameters, err = prepStartWorkspace(inv, client, prepStartWorkspaceArgs{
+					Action:            WorkspaceStart,
+					TemplateVersionID: template.ActiveVersionID,
+
+					LastBuildParameters: lastBuildParameters,
+
+					PromptBuildOptions: parameterFlags.promptBuildOptions,
+					BuildOptions:       buildOptions,
+				})
+				if err != nil {
+					return err
+				}
+
+				req.RichParameterValues = buildParameters
+				req.TemplateVersionID = template.ActiveVersionID
+				build, err = client.CreateWorkspaceBuild(inv.Context(), workspace.ID, req)
+				if err != nil {
+					return err
+				}
```

Comment:
worth extracting this logic into a function and calling it from restart.go? i.e. don't repeat yourself

Reply:
idk why CI doesn't catch this in the `fmt` job 🤔